### ISSUE TITLE
Installation on fedora-clang/macOS with clang 14 and C++17

### ISSUE THREAD

**Issue by sneumann** [girke-lab/ChemmineR#13]
Hi, as reported by Prof. Brian Ripley:
_These do not install on fedora-clang and macOS with Apple clang 14 (at least) with R's new default of C++17. Log extracts below.
C++ has never supported 'register' (except in system headers), and all the removed functions were deprecated as long ago as C++11.
You could declare a C++ standard (see 'Writing R Extensions') but it would be better to modernize your code._
```
r_wrap.cc:563:7: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
       register size_t l = 0;
       ^~~~~~~~~
r_wrap.cc:564:7: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
       register size_t r = iter->size - 1;
       ^~~~~~~~~
r_wrap.cc:567:2: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
         register size_t i = (l + r) >> 1;
         ^~~~~~~~~
r_wrap.cc:570:4: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
           register int compare = strcmp(name, iname);
           ^~~~~~~~~
r_wrap.cc:614:7: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
       register size_t i = 0;
       ^~~~~~~~~
r_wrap.cc:633:3: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
   register const unsigned char *u = (unsigned char *) ptr;
   ^~~~~~~~~
r_wrap.cc:634:3: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
   register const unsigned char *eu =  u + sz;
   ^~~~~~~~~
r_wrap.cc:636:5: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
     register unsigned char uu = *u;
     ^~~~~~~~~
r_wrap.cc:648:3: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
   register unsigned char *u = (unsigned char *) ptr;
   ^~~~~~~~~
r_wrap.cc:649:3: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
   register const unsigned char *eu = u + sz;
   ^~~~~~~~~
r_wrap.cc:651:5: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
     register char d = *(c++);
     ^~~~~~~~~
r_wrap.cc:652:5: error: ISO C++17 does not allow 'register' storage 
class specifier [-Wregister]
     register unsigned char uu;
     ^~~~~~~~~
```
In e.g. https://github.com/girke-lab/ChemmineR/blob/master/src/r_wrap.cc#L563
Yours, Steffen

**Comment by khoran** [comment]
That code was generated by an old version of SWIG (2.x). I'm working on re-generating it with SWIG 4.x, but it's giving me some runtime errors. I'll let you know when I get it working.

**Comment by khoran** [comment]
Fixed in dev version 3.51.1
